test/Guzzle/SubjectRewritingMiddlewareTest.php

```diff
@@ -0,0 +1,693 @@
+                        ],
+                    ],
+                ],
+            ],
+            'application/vnd.elife.highlight-list+json; version=1' => [
+                [
+                    'total' => 1,
+                    'items' => [
+                        [
+                            'title' => 'Highlight title',
+                            'item' => $this->createArticlePoA(false),
+                        ],
+                    ],
+                ],
+                [
+                    'total' => 1,
+                    'items' => [
+                        [
+                            'title' => 'Highlight title',
+                            'item' => $this->createArticlePoA(true),
+                        ],
+                    ],
+                ],
+            ],
+            'application/vnd.elife.person+json; version=1' => [
+                $this->createPerson(false, false),
+                $this->createPerson(true, false),
+            ],
+            'application/vnd.elife.person-list+json; version=1' => [
+                [
+                    'total' => 1,
+                    'items' => [
+                        $this->createPerson(false, false),
+                    ],
+                ],
+                [
+                    'total' => 1,
+                    'items' => [
+                        $this->createPerson(true, false),
+                    ],
+                ],
+            ],
+            'application/vnd.elife.podcast-episode+json; version=1' => [
+                $this->createPodcastEpisode(false, false),
+                $this->createPodcastEpisode(true, false),
+            ],
+            'application/vnd.elife.press-package+json; version=3' => [
+                $this->createPressPackage(false, false),
+                $this->createPressPackage(true, false),
+            ],
+            'application/vnd.elife.press-package-list+json; version=1' => [
+                [
+                    'total' => 1,
+                    'items' => [
+                        $this->createPressPackage(false),
+                    ],
+                ],
+                [
+                    'total' => 1,
+                    'items' => [
+                        $this->createPressPackage(true),
+                    ],
+                ],
+            ],
+            'application/vnd.elife.recommendations+json; version=1' => [
+                [
+                    'total' => 2,
+                    'items' => [
+                        $this->createArticlePoA(false),
+                        $this->createArticleVoR(false),
+                    ],
+                ],
+                [
+                    'total' => 2,
+                    'items' => [
+                        $this->createArticlePoA(true),
+                        $this->createArticleVoR(true),
+                    ],
+                ],
+            ],
+            'application/vnd.elife.search+json; version=1' => [
```

Comment:
test seems alright, although there are many overlapping `2` values to cover different concepts. If it were starting from `3` in `new-subject` and `7` in `other-subject`, it would make movements to the wrong place more obvious

Reply:
Updated.